### PR TITLE
feat: add C `ndarray` API and refactor `blas/ext/base/dsnansum`

### PR DESCRIPTION
**Pull request by headlessNode**
## Description

> What is the purpose of this pull request?

This pull request:

-   add C `ndarray` API and refactor/cleanup `blas/ext/base/dsnansumpw`

## Related Issues

> Does this pull request have any related issues?

No. 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/benchmark/benchmark.js [hunk 1]
```diff
@@ -22,7 +22,7 @@
 
 var bench = require( '@stdlib/bench' );
 var bernoulli = require( '@stdlib/random/base/bernoulli' );
-var discreteUniform = require( '@stdlib/random/base/discrete-uniform' );
+var uniform = require( '@stdlib/random/base/uniform' );
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/benchmark/benchmark.js:25]: @kgryte This change is based on how random numbers are generated in other similar packages i.e. dsnansumors and dsnansumpw.

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/include/stdlib/blas/ext/base/dsnansum.h [hunk 2]
```diff
@@ -31,7 +31,12 @@ extern "C" {
 /**
 * Computes the sum of single-precision floating-point strided array elements, ignoring `NaN` values, using extended summation, and returning an extended precision result.
 */
-double stdlib_strided_dsnansum( const int64_t N, const float *X, const int64_t stride );
+double API_SUFFIX(stdlib_strided_dsnansum)( const CBLAS_INT N, const float *X, const CBLAS_INT strideX );
+
+/**
+* Computes the sum of single-precision floating-point strided array elements, ignoring `NaN` values, using extended summation and alternative indexing semantics, and returning an extended precision result.
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/include/stdlib/blas/ext/base/dsnansum.h]:
```suggestion
* Computes the sum of single-precision floating-point strided array elements, ignoring `NaN` values, using extended accumulation and alternative indexing semantics, and returning an extended precision result.
```

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/include/stdlib/blas/ext/base/dsnansum.h [hunk 3]
```diff
@@ -31,7 +31,12 @@ extern "C" {
 /**
 * Computes the sum of single-precision floating-point strided array elements, ignoring `NaN` values, using extended summation, and returning an extended precision result.
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/include/stdlib/blas/ext/base/dsnansum.h]:
```suggestion
* Computes the sum of single-precision floating-point strided array elements, ignoring `NaN` values, using extended accumulation, and returning an extended precision result.
```

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/benchmark/benchmark.js (hunk 1 above)
Comment (kgryte) [reply lib/node_modules/@stdlib/blas/ext/base/dsnansum/benchmark/benchmark.js:25]: Seems reasonable. Thanks for the context!

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/README.md [hunk 4]
```diff
@@ -136,14 +135,14 @@ var discreteUniform = require( '@stdlib/random/base/discrete-uniform' );
...
-function clbk() {
+function rand() {
     if ( bernoulli( 0.7 ) > 0 ) {
         return discreteUniform( -10, 10 );
     }
     return NaN;
 }
 
-var x = filledarrayBy( 10, 'float64', clbk );
+var x = filledarrayBy( 10, 'float32', rand );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/README.md:145]: Thanks for catching this dtype mistake!

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @headlessNode!

**Pull request merged**